### PR TITLE
Gradient accumulation

### PR DESCRIPTION
**Pull request by RRoundTable**
noisy gradient에 대한 글을 썼습니다.

내용은 아래와 같습니다.
- residual ae가 gradient가 noisy한 이유
- gradient accumulation이 무엇인지
- result

### CODE REVIEW

**Comment by kh-kim** [comment]
- Noisy Gradient Problem에 대해서 좀 더 자세한 정의/설명이 있으면 좋겠습니다. 너무 그림만 던져놓은 느낌.
  - e.g. Stochastic Gradient Descent (SGD)를 활용한 경우, 일반적으로 미니배치의 gradient는 전체 데이터셋으로부터 구한 gradient에 비해서 오차(variance)가 존재할 수 있습니다. 이때 이 variance가 큰 경우 gradient가 noisy하다고 볼 수 있고, 이는 최적화를 수행할 때 어려움으로 작용할 수 있습니다. 이 포스트에서는 이와 같은 문제를 noisy gradient problem이라고 부르도록 하겠습니다.
  - 기존에 Transformer와 관련된 연구들에서 이와 같은 이슈를 다루었다고 ref 표기를 해주면 더 신뢰를 줄 수 있습니다.
- 4개 레이어의 일반 forward & residual forward 예제를 수식으로 제대로 표현하던지, 아니면 그림으로 그리는게 낫겠습니다.
  - 네트워크에는 \ell 개의 레이어가 존재한다고 합시다.
  - 각 레이어를 f_i 라고 표현하고, 중간 결과물을 h_i라고 표현하며, h_0=x, h_\ell=\hat{y}, loss=||x-\hat{y}||_2^2 라고 표현하심이 어떤지요?
  - 그럼 h_i=f_i(h_{i-1})이 됩니다.
  - skip-connection의 경우에는 h_i=f_i(h_{i-1})+h_{i-1}
  - 그럼 \hat{y}=h_\ell=f_i(h_{i-1})+f_{i-1}(h_{i-2})+...=\sum_{i=1}^\ell{f_i(h_{i-1})} 이라고 표현 가능
- Variance 표현은 좋음
- "이런 문제를 해결하기 위해서 크게 3가지 정도의 시도" 에서 reference 표기
- 모멘텀을 사용하지 않는 옵티마이저 --> 예시 필요 e.g. AdaDelta, RMSProp + ref 추가
- LR Warm up & Decay 에도 ref를 추가
- RAdam 언급 및 ref 추가
- Residual VAE vs VAE --> 섹션 이름을 Evaluation 으로 변경
- 실험 셋팅을 좀 더 자세히 명시
  - 예를 들어 0이 target class일때는 1부터 9는 정상 클래스로 학습 데이터에 포함되었으며,, 어쩌고 저쩌고
- Conclusion에 gardient accumulation의 효과와 원인을 다시 한번 더 언급

**Comment by RRoundTable** [comment]
@kh-kim  comment 감사드립니다.

**Comment by RRoundTable** [comment]
- [x] Noisy Gradient Problem에 대해서 좀 더 자세한 정의/설명이 있으면 좋겠습니다. 너무 그림만 던져놓은 느낌.
- [x] 기존에 Transformer와 관련된 연구들에서 이와 같은 이슈를 다루었다고 ref 표기를 해주면 더 신뢰를 줄 수 있습니다.
- [] 4개 레이어의 일반 forward & residual forward 예제를 수식으로 제대로 표현하던지, 아니면 그림으로 그리는게 낫겠습니다.
- [x] "이런 문제를 해결하기 위해서 크게 3가지 정도의 시도" 에서 reference 표기
- [x] 모멘텀을 사용하지 않는 옵티마이저 --> 예시 필요 e.g. AdaDelta, RMSProp + ref 추가
- [ ] LR Warm up & Decay 에도 ref를 추가
- [ ] RAdam 언급 및 ref 추가
- [x] Residual VAE vs VAE --> 섹션 이름을 Evaluation 으로 변경
- [x] 실험 셋팅을 좀 더 자세히 명시
- [x] Conclusion에 gardient accumulation의 효과와 원인을 다시 한번 더 언급
